EVO-related issues with satvalue and slider

Manarnz opened the pull request:
This closes issue #221
- [x] Before, when switching back to off or modeling mode, the satvalue does not update to the initial sat val. Instead it is the last satvalue checked from the slider
- [x] Before, when simulating path, the slider would pop up for all EVO options. Now the slider shows only in state sliderOption
- [x] Before, the satvlaue showed up for both `%` and `time` sliderOption, now it does not
- [x] Before, the text color in modeling mode would not update back to black after simulating path
- [x] Before, when switching configs, the analysis slider would not update to modeling slider
- [x] Fixed issue with `revertNodeValuesToInitial()`

Manarnz committed:
fixed issue #221

Manarnz committed:
Removed unnecessary method

I created the update method in the last PR and forgot to delete it since we don't call it in the code.

Manarnz committed:
Merge branch 'backbone-refactor' of https://github.com/amgrubb/BloomingLeaf into ma-issue221

Manarnz committed:
Fixed merge problems

Manarnz committed:
Fixed issue with select and slider

Manarnz committed:
Fixed issue with satvalue and revertNodeValuesToInitial

Manarnz committed:
Merge branch 'backbone-refactor' of https://github.com/amgrubb/BloomingLeaf into ma-issue221

Manarnz committed:
Fixed issue with EVO coloring mode when creating new config

Before, when creating a new config, the EVO goes off but the EVO mode view is still on.

Manarnz committed:
Fixed issue with EVO coloring

Before, when switching from one config to the other, the color mode would be for the sliderOption before the selected sliderOption. So if it is at "satae" it shows the oloring system for "%"" etc

YesugenB commented:
When you're in Analysis View and simulate single path, change to the `time` sliderOption click Modeling View, and then click "No return to analysis view", the sliderOption changes to `%` 
Is this the intended behavior?

YesugenB commented on leaf-ui/js/object/evoObjects.js:
```diff
@@ -167,7 +166,9 @@ class EVO {
-                EVO.revertIntentionsText(graph.getElements(), paper);    
+                EVO.revertIntentionsText(graph.getElements(), paper);
+                EVO.displaySlider(false);
+
```
Can delete this line and then fix the indentation for the` break; `below

YesugenB commented on leaf-ui/js/onFunctions.js:
```diff
@@ -741,31 +739,31 @@ function setInteraction(interactionValue) {
+    	} else {
+            curr.attr('.satvalue/text', satisfactionValuesDict[initSatVal].satValue);
+    	}
+        //curr.attr({text: {fill: 'black'}});
```
Can this be removed?

YesugenB commented on leaf-ui/js/onFunctions.js:
```diff
@@ -741,31 +739,31 @@ function setInteraction(interactionValue) {
+    		continue;
+    	}     
+    	var intention = curr.get('intention');
+        var initSatVal = intention.getUserEvaluationBBM(0).get('assignedEvidencePair'); 
```
Fix the indentation and then delete extra line below

YesugenB commented on leaf-ui/rappid-extensions/ConfigInspector.js:
```diff
@@ -142,6 +142,8 @@ var Config = Backbone.View.extend({
         this.$('.analysis-configuration').append(this.innerView.$el);
         if (this.model.get('selected')) {
             this.showAnalysisInspector();
+            //this.resetAnalysis();
```
Can this be removed?

Manarnz commented:
> When you're in Analysis View and simulate single path, change to the `time` sliderOption click Modeling View, and then click "No return to analysis view", the sliderOption changes to `%`
> Is this the intended behavior?

No it is not. Cat in her PR `Fixed EVO Coloring Bug for State #526` fixed this issue. I believe.

Manarnz committed:
Implemented Yesugen's comments and deleted chnages to `revertNodeValuesToInitial()`

Domily11 committed:
Fixed EVO Coloring Bug for State (#526)

* Fixed #525

* Responded to comments

* Reimplemented revertNodeValuesToInitial()

* Integrated IntentionColoring into EVO

* Add some docstring

* Implemented comments

Co-authored-by: Manarnz <83615524+Manarnz@users.noreply.github.com>
Co-authored-by: A. M. Grubb <amgrubb@smith.edu>

Manarnz commented:
Bugs:
- [x] When switching between results, the slider is at timePoint 0 but the EVO color is at the last timePoint
- [x] When turning EVO off while no result is selected and then selecting a result, the EVO is set at sliderOption 1 but the EVO color is off

Manarnz committed:
Fixed second bug in the comment

Created a refreshSlider method to fix the issue for when the coloring is not updated while switching between config and request. Additionaly, I removed unnecessary refresh and the method EVO.update

Manarnz committed:
Cleaned code and added comments

i-m-foster reviewed (approved):
Besides adding the slider if EVO is off it looks good to me!

Manarnz committed:
Made sliderShow when result is clicked and sliderOption is off

amgrubb reviewed (approved):
Nice work, I'm glad we merged these first. Please squash and merge.
